### PR TITLE
Adding in the repo flag to the regenerate target in the plugin make t…

### PR DESCRIPTION
**Pull request by StabbyCutyou**
…emplate

ping @rossnanop @jgkomand 

I'm still working on this, and am gonna see about a sister PR into komand to make it so the default for the flag is sane so that there is less chance something goes wrong, but essentially:

The regen call is missing the now-required repo flag, which effectively becomes the PackageRoot value in the templates - which in a plugin like our slack plugin, should be `github.com/komand/plugins/slack`.

### CODE REVIEW

**Comment by StabbyCutyou** [comment]
ping @jandre WIP changes to logging, debug flag.

ping @rossnanop you're already using this, so be aware of these logging changes - essentially Printf and Println are going away in favor of a dirt simple leveled logger. YES, I KNOW ABOUT THE ARDAN LABS LOGGER. I didn't do that because i wanted the "Normal" and "Buffered" loggers to look the same, and I didn't wanna write some de-muxer for multiple bytes.buffers, i'm doing this this way in the interest of time.

**Comment by StabbyCutyou** [comment]
ping @rossnanop @jandre if you guys have time to provide feedback, that'd be good.

You can see the results of these changes in the slack plugin pr

File: templates/Makefile.template [hunk 1]
```diff
@@ -32,6 +32,10 @@ icon:
 	sed -i.bu "/^icon/s/\(icon: \).*/\1\"data:image\/png;base64,$(ENCODING)\"/" plugin.spec.yaml || true
 	@rm -f *.bu || true
 
+tarball:
+	rm -rf $(PKG)
+	tar -cvzf $(PKG) --exclude=$(PKG) *
+
```
Comment (StabbyCutyou) [inline templates/Makefile.template]: ping @jonschipp @jcsalem @jandre from the other pr

File: templates/Makefile.template [hunk 2]
```diff
@@ -78,4 +82,8 @@ check:
 	gosimple $$(go list ./... | grep -v /vendor/)
 	unused $$(go list ./... | grep -v /vendor/)
 
+export: image
+	@printf "\n ---> Exporting Docker image to ./$(NAME).img\n"
+	@docker save $(VENDOR)/$(NAME):$(VERSION) | gzip > $(NAME)_$(VERSION).img
+
```
Comment (StabbyCutyou) [inline templates/Makefile.template]: ping @jonschipp @jcsalem @jandre from the other pr

**Comment by StabbyCutyou** [comment]
Specifically, i got the makefile changes from:

https://github.com/komand/komand/commit/7388861981e2c23a25709b0356d84d5459e77ec1

**Comment by StabbyCutyou** [comment]
Also, i see now where moving the makefile (and possible the dockerfile) out of single location is going to be a big pain in the butt.

I will open a PR to fix this in komand/komand as soon as I can :)

**Comment by StabbyCutyou** [comment]
some one look at meeeeeeeee

**Comment by StabbyCutyou** [comment]
Need to get this merged in to unblock some other things. ping @jandre @rossnanop if i can get some feedback or an LGTM today, that'd be great.

File: templates/cmd/main.template [hunk 3]
```diff
@@ -279,7 +285,7 @@ func printInfo() {
 	result += fmt.Sprintf("\n")
 	result += fmt.Sprintf("Triggers (%s%d%s): \n", cli.Green, {{- $trigcount }}, cli.Reset)
 	{{- range $i, $t := .Triggers }}
-	result += fmt.Sprintf("└── %s%s%s (%s%s)\n", cli.Green, "{{- $t.Name }}", cli.Reset, "{{- $t.Description }}", cli.Reset)
+	result += fmt.Sprintf("└── %s%s%s (%s%s)\n", cli.Green, "{{- $t.RawName }}", cli.Reset, "{{- $t.Description }}", cli.Reset)
```
Comment (StabbyCutyou) [inline templates/cmd/main.template]: ping @jonschipp

File: templates/cmd/main.template [hunk 4]
```diff
@@ -289,7 +295,7 @@ func printInfo() {
 	result += fmt.Sprintf("\n")
 	result += fmt.Sprintf("Actions (%s%d%s): \n", cli.Green, {{ $actioncount }}, cli.Reset)
 	{{- range $i, $a := .Actions }}
-	result += fmt.Sprintf("└── %s%s%s (%s%s)\n", cli.Green, "{{- $a.Name }}", cli.Reset, "{{- $a.Description }}", cli.Reset)
+	result += fmt.Sprintf("└── %s%s%s (%s%s)\n", cli.Green, "{{- $a.RawName }}", cli.Reset, "{{- $a.Description }}", cli.Reset)
```
Comment (StabbyCutyou) [inline templates/cmd/main.template]: and @jonschipp again

**Comment by rossnanop** [comment]
@StabbyCutyou LGTM with one question do the plugins I merged in need to be updated to use this logger?

**Comment by StabbyCutyou** [comment]
100% yes.

But since the plugins vendor the SDK, you shouldn't have to rush.

------------------------------
 *Sean Kelly*
Principal Software Engineer
*p:* 339-927-4452     *e:* sean@komand.com
[image: Komand] <https://www.komand.com>

**Pull request merged**